### PR TITLE
Add candidate lifecycle with pruning and position filtering

### LINKED ISSUE #356: Candidates have no lifecycle — remain in pipeline indefinitely
## Problem

Once a ticker is scanned and added to the candidates table, it remains in the pipeline forever. There is no mechanism to remove or expire candidates based on their outcome or the state of the market.

No `DELETE` or cleanup logic exists for the candidates table anywhere in the codebase.

## Impact

- Candidates that were opened as positions still appear in the pipeline
- Candidates that were explicitly rejected/skipped still appear
- Candidates for finalized/closed markets still appear
- The pipeline grows unboundedly over time

## Expected Behavior

Candidates should have a lifecycle with clear exit conditions, such as:
- Opened as a position
- Explicitly rejected/skipped
- Market finalized or closed
- Aged out after N cycles without action

The Candidate Pipeline panel should reflect the current state of actionable candidates, not a historical record of everything ever scanned.

### PR DESCRIPTION
**Pull request by allan-mobley-jr**
## Summary
- Adds `prune_candidates()` function with four pruning rules: opened as position, market inactive, aged out (72h default), stale duplicates — all executed atomically in a single transaction
- Adds hidden `prune-candidates` CLI command with optional `--check-markets` flag for live API market status checks
- Updates `get_candidates()` dashboard query to exclude candidates whose ticker has an open position (LEFT JOIN anti-pattern)
- Fixes `get_position_tickers()` to be mode-aware — was hardcoded to `positions` table, missing `paper_positions` in driving range mode
- Fixes `KalshiClient` resource leak in market-checking path (now uses `async with`)
- Adds `position_table` property to `GimmesConfig` to centralize table name logic

Closes #356

## Test plan
- [x] 18 new tests in `test_prune_candidates.py` covering all pruning rules, edge cases, combined rules, overlap behavior, and input validation
- [x] 3 new tests in `test_clubhouse.py` for position filtering, settled-position inclusion, and pagination with filtering
- [x] Full test suite passes (836 tests)
- [x] Lint clean on all changed files

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/gimmes/clubhouse/data.py [hunk 1]
```diff
@@ -291,10 +297,13 @@ async def get_candidates(
                 " INNER JOIN ("
                 "   SELECT MAX(id) AS max_id FROM candidates GROUP BY ticker"
                 " ) latest ON c.id = latest.max_id"
+                f" LEFT JOIN {pos_table} p"
+                " ON c.ticker = p.ticker AND p.count > 0"
+                " WHERE p.ticker IS NULL"
             )
```
[the remark this answers is not in the document]
Comment (allan-mobley-jr) [reply src/gimmes/clubhouse/data.py]: This is handled correctly — the pre-existing `except Exception` block at line 325 already catches `OperationalError` from a missing table and returns an empty list with a warning log. The `paper_positions` table is created by the paper broker's `_ensure_schema` which runs before any dashboard query in normal operation. The missing-table scenario is an edge case that degrades gracefully.
